com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/DefoldLuaParser.java

```diff
@@ -0,0 +1,189 @@
+package com.dynamo.bob.pipeline;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import org.antlr.v4.runtime.CommonTokenStream;
+import org.antlr.v4.runtime.tree.ParseTree;
+import org.antlr.v4.runtime.tree.ParseTreeWalker;
+import org.antlr.v4.runtime.CharStreams;
+import org.antlr.v4.runtime.CharStream;
+import org.antlr.v4.runtime.ParserRuleContext;
+import org.antlr.v4.runtime.Token;
+
+import com.dynamo.bob.pipeline.luaparser.LuaParser;
+import com.dynamo.bob.pipeline.luaparser.LuaLexer;
+import com.dynamo.bob.pipeline.luaparser.LuaParserBaseListener;
+
+
+public class DefoldLuaParser extends LuaParserBaseListener {
```

Comment:
This is the new Lua parser. It uses callbacks from the generated parser code from ANTLR to detect function calls and process the entire structure of a Lua file.